plugins/org.locationtech.udig.ui/src/org/locationtech/udig/ui/CRSChooser.java

```diff
@@ -475,7 +475,10 @@ protected CoordinateReferenceSystem createCRS( String code ) {
         for( Object object : ReferencingFactoryFinder.getCRSAuthorityFactories(null) ) {
             CRSAuthorityFactory factory = (CRSAuthorityFactory) object;
             try {
-                return (CoordinateReferenceSystem) factory.createObject(code);
+                IdentifiedObject identifiedObject = factory.createObject(code);
+                if (identifiedObject instanceof CoordinateReferenceSystem) {
+                    return (CoordinateReferenceSystem) factory.createObject(code);
+                }
             } catch (FactoryException e2) {
```

Comment:
Not sure about a better way, I tend to use the CRS.decode( string ) method. I think the internals of that method would be a good way to see how these factories are used.

However this instance of sanity check seems fine...